edbdeploy/data/terraform/azure-db/environments/vm/vm.tf

```diff
@@ -0,0 +1,267 @@
+
+  identity {
+    type = "SystemAssigned"
+  }
+
+  source_image_reference {
+    publisher = var.azure_publisher
+    offer     = var.azure_offer
+    sku       = var.azure_sku
+
+    version = "latest"
+  }
+
+  os_disk {
+    name                 = format("pg-%s-%s-%s", var.cluster_name, "EDB-VM-OS-Disk", count.index)
+    storage_account_type = var.postgres_server["volume"]["storage_account_type"]
+    caching              = "ReadWrite"
+  }
+
+  additional_capabilities {
+    ultra_ssd_enabled = true
+  }
+
+  tags = var.project_tags
+}
+
+resource "azurerm_linux_virtual_machine" "hammerdb_server" {
+  count               = var.hammerdb_server["count"]
+  name                = format("%s-%s%s", var.cluster_name, "hammerdbserver", count.index + 1)
+  resource_group_name = var.resourcegroup_name
+  location            = var.azure_region
+
+  size           = var.hammerdb_server["instance_type"]
+  admin_username = var.ssh_user
+
+  network_interface_ids = [element(azurerm_network_interface.hammerdb_public_nic.*.id, count.index)]
+
+  admin_ssh_key {
+    username   = var.ssh_user
+    public_key = file(var.ssh_pub_key)
+  }
+
+  identity {
+    type = "SystemAssigned"
+  }
+
+  source_image_reference {
+    publisher = var.azure_publisher
+    offer     = var.azure_offer
+    sku       = var.azure_sku
+
+    version = "latest"
+  }
+
+  os_disk {
+    name                 = format("hammerdb-%s-%s-%s", var.cluster_name, "EDB-VM-OS-Disk", count.index)
+    storage_account_type = var.hammerdb_server["volume"]["storage_account_type"]
+    caching              = "ReadWrite"
+  }
+
+  tags = var.project_tags
+}
+
+resource "azurerm_linux_virtual_machine" "pem_server" {
+  count               = var.pem_server["count"]
+  name                = format("%s-%s%s", var.cluster_name, "pemserver", count.index + 1)
+  resource_group_name = var.resourcegroup_name
+  location            = var.azure_region
+
+  size           = var.pem_server["instance_type"]
+  admin_username = var.ssh_user
+
+  network_interface_ids = [element(azurerm_network_interface.pem_public_nic.*.id, count.index)]
+
+  admin_ssh_key {
+    username   = var.ssh_user
+    public_key = file(var.ssh_pub_key)
+  }
+
+  identity {
+    type = "SystemAssigned"
+  }
+
+  source_image_reference {
+    publisher = var.azure_publisher
+    offer     = var.azure_offer
+    sku       = var.azure_sku
+
+    version = "latest"
+  }
+
+  os_disk {
+    name                 = format("pem-%s-%s-%s", var.cluster_name, "EDB-VM-OS-Disk", count.index)
+    storage_account_type = var.pem_server["volume"]["storage_account_type"]
+    caching              = "ReadWrite"
+  }
+
+  tags = var.project_tags
+}
+
+resource "azurerm_postgresql_server" "postgresql_server" {
+  name = format("%s-server", var.cluster_name)
+  location = var.azure_region
+  resource_group_name = var.resourcegroup_name
+ 
+  administrator_login          = "postgres"
+  administrator_login_password = "Ahb0uL!eerahmiT4ShooF6teepie2eu6"
```

Comment:
There is absolutely no way to pass the password from the python code? Having hardcoded passwords makes me nervous.

Reply:
Oops, that was on my TODO.  Will work on it.